webapp/channels/src/components/admin_console/feature_discovery/feature_discovery.test.tsx

```diff
@@ -2,103 +2,189 @@
-                    copyDefault={'Bar'}
-                    learnMoreURL='https://test.mattermost.com/secondary/'
-                    featureDiscoveryImage={<SamlSVG/>}
-                    // eslint-disable-next-line @typescript-eslint/naming-convention
-                    stats={{TOTAL_USERS: 20}}
-                    prevTrialLicense={{IsLicensed: 'false'}}
-                    isCloud={true}
-                    isCloudTrial={false}
-                    hadPrevCloudTrial={false}
-                    isSubscriptionLoaded={false}
-                    cloudFreeDeprecated={false}
-                    isPaidSubscription={false}
-                    actions={{
-                        getPrevTrialLicense: jest.fn(),
-                        openModal: jest.fn(),
-                        getCloudSubscription: jest.fn(),
-                    }}
-                />,
+        test('should match component state when is cloud environment and subscription is not loaded yet in redux store', () => {
+            const getPrevTrialLicense = jest.fn();
+            const getCloudSubscription = jest.fn();
+            const openModal = jest.fn();
+
+            renderWithIntl(
+                <ReduxProvider store={store}>
+                    <FeatureDiscovery
+                        featureName='test'
+                        minimumSKURequiredForFeature={LicenseSkus.Professional}
+                        titleID='translation.test.title'
+                        titleDefault='Foo'
+                        copyID='translation.test.copy'
+                        copyDefault={'Bar'}
+                        learnMoreURL='https://test.mattermost.com/secondary/'
+                        featureDiscoveryImage={<SamlSVG/>}
+                        // eslint-disable-next-line @typescript-eslint/naming-convention
+                        stats={{TOTAL_USERS: 20}}
+                        prevTrialLicense={{IsLicensed: 'false'}}
+                        isCloud={true}
+                        isCloudTrial={false}
+                        hadPrevCloudTrial={false}
+                        isSubscriptionLoaded={false}
+                        isPaidSubscription={false}
+                        cloudFreeDeprecated={false}
+                        actions={{
+                            getPrevTrialLicense,
+                            getCloudSubscription,
+                            openModal,
+                        }}
+                    />
+                </ReduxProvider>,
             );
-            expect(wrapper).toMatchSnapshot();
+
+            // when is cloud and subscription is not loaded yet, then only loading spinner is visible
+            expect(screen.getByTestId('loadingSpinner')).toBeInTheDocument();
+
+            expect(screen.queryByText('Bar')).not.toBeInTheDocument();
+            expect(screen.queryByText('Foo')).not.toBeInTheDocument();
+
+            //this option is visible only when it is cloud environment
```

Comment:
I wonder if this comment is correct. The comment says "visible" but the code is "not.toBeInTheDocument".

Reply:
Yeah make sense I will change the wording